core/templates/dev/head/services/StateTopAnswersStatsService.js

```diff
@@ -52,21 +55,23 @@ oppia.factory('StateTopAnswersStatsService', [
        * Calls the backend asynchronously to setup the answer statistics of each
        * state this exploration contains.
        */
-      init: function() {
-        $http.get(
-          UrlInterpolationService.interpolateUrl(
-            '/createhandler/state_answer_stats/<exploration_id>',
-            {exploration_id: ExplorationContextService.getExplorationId()})
-        ).then(function(response) {
-          stateTopAnswerStatsCache = {};
-          Object.keys(response.data.answers).forEach(function(stateName) {
-            var answerStatsBackendDicts = response.data.answers[stateName];
-            stateTopAnswerStatsCache[stateName] = answerStatsBackendDicts.map(
-              AnswerStatsFactory.createFromBackendDict);
-            // Still need to manually refresh the addressed information.
-            refreshAddressedInfo(stateName);
-          });
+      init: function(stateTopAnswersStatsBackendDict) {
+        stateTopAnswerStatsCache = {};
+        Object.keys(
```

Comment:
Actually, I'm only seeing it now, but this is a bit of a strange construct. It creates a copy of the keys of the object and then iterates through them with forEach. Why not just do "for stateName in stateTopAnswersStatsBackendDict.answers"?

Reply:
Ah, I thought those didn't work well in JavaScript, fixed to use that!